src/appleseed.shaders/src/appleseed/as_manifold2d.osl

```diff
@@ -0,0 +1,497 @@
+    string as_max_class_id = "1185226119 622806922",
+    string as_max_plugin_type = "texture",
+    string URL = "https://appleseed.readthedocs.io/projects/appleseed-maya/en/latest/shaders/utilities/as_manifold2D.html#label-as-manifold2D"
+]]
+(
+    float in_uv_coord[2] = {u, v}
+    [[
+        string as_maya_attribute_name = "uvCoord",
+        string label = "UV Coordinates",
+        string page = "Coordinates",
+        int as_max_attribute_hidden = 1
+    ]],
+    float in_uv_filter_size[2] = {
+        UNDEFINED_UVFILTER, UNDEFINED_UVFILTER}
+    [[
+        string as_maya_attribute_name = "uvFilterSize",
+        string as_maya_attribute_short_name = "uvf",
+        string label = "UV Filter Size",
+        string page = "Coordinates",
+        string help = "UV coordinates filter widths",
+        int divider = 1
+    ]],
+    int in_bypass_uv = 0
+    [[
+        string as_maya_attribute_name = "bypassUV",
+        string as_maya_attribute_short_name = "buv",
+        string widget = "checkBox",
+        string label = "Bypass UV transforms",
+        string page = "Transforms",
+        int as_maya_attribute_connectable = 0,
+        int as_maya_attribute_keyable = 0,
+        int as_maya_attribute_hidden = 1,
+        int as_blender_input_socket = 0,
+        int gafferNoduleLayoutVisible = 0,
+        int divider = 1
+    ]],
+    int in_compute_filters = 1
+    [[
+        string as_maya_attribute_name = "computeFilters",
+        string as_maya_attribute_short_name = "cfi",
+        string widget = "checkBox",
+        string label = "Compute UV Filters",
+        string page = "Transforms",
+        int as_maya_attribute_connectable = 0,
+        int as_maya_attribute_keyable = 0,
+        int as_maya_attribute_hidden = 1,
+        int as_blender_input_socket = 0,
+        int gafferNoduleLayoutVisible = 0,
+        string help = "Compute the filter widths of the UV coordinates.",
+        int divider = 1
+    ]],
+
+    //
+    // UV frame
+    float in_scale_frame[2] = {1.0, 1.0}
+    [[
+        string as_maya_attribute_name = "scaleFrame",
+        string as_maya_attribute_short_name = "sfr",
+        string label = "Scale UV Frame",
+        string page = "Frame",
+        string help = "Scale the UV frame."
+    ]],
+    float in_translate_frame[2] = {0.0, 0.0}
+    [[
+        string as_maya_attribute_name = "translateFrame",
+        string as_maya_attribute_short_name = "tfr",
+        string label = "Translate UV Frame",
+        string page = "Frame",
+        string help = "Translate the UV frame."
+    ]],
+    float in_rotate_frame = 0.0
+    [[
+        string as_maya_attribute_name = "rotateFrame",
+        string as_maya_attribute_short_name = "rfr",
+        float softmin = 0.0,
+        float softmax = 1.0,
+        string label = "Rotate UV Frame",
+        string page = "Frame",
+        string help = "Rotate the UV frame, where values in the open interval (0,1) correspond to (0,360) degrees."
```

Comment:
Right now it's soft-restricted, so in the UI it appears as a slider between 0 and 1, but Maya as a numeric field besides the slider, so if you want to go outside the range, you can enter a number there and the slider rescales itself into the new range. It's there mostly to allow using expressions, custom/dynamic attributes in Maya, in case they need negative values.